python_modules/dagster/dagster/_core/storage/upath_io_manager.py

```diff
@@ -425,6 +425,18 @@ def load_input(self, context: InputContext) -> Union[Any, Dict[str, Any]]:
 
                 return self._load_multiple_inputs(context)
 
+    def _handle_transition_to_partitioned_asset(self, context: OutputContext, path: "UPath"):
+        # if the asset was previously non-partitioned, path will be a file, when it should
+        # be a directory for the partitioned asset. Delete the file, so that it can be made
+        # into a directory
+        if path.exists() and path.is_file():
+            context.log.warn(
+                f"Non-partitioned asset {context.asset_key} was materialized to"
+                " file {path}. Materializing {context.asset_key} as a partitioned asset,"
+                f" will delete this file, and create a directory, {path} containing partitioned data files."
```

Comment:
the tense here reads a little weird to me, this is going to happen so maybe change it up. What pops in my head is something like: 

"Found file at {path} believed to correspond with previously non-partitioned version of {asset}. Removing file and replacing with directory for partitioned data files"